src/main/java/org/wso2/carbon/esb/connector/SessionManager.java

```diff
@@ -60,7 +60,11 @@ private String getKey(String host, int port, String systemId) {
     public SMPPSession getSmppSession(int enquireLinkTimer, int transactionTimer, String host, int port,
                                       BindParameter bindParameter) throws IOException {
         SMPPSession smppSession = smppSessionList.get(getKey(host, port, bindParameter.getSystemId()));
-        if (smppSession == null) {
+        Boolean isSessionBound=false;
+        if(smppSession != null){
+            isSessionBound=smppSession.getSessionState().isBound();
+        }
+        if (!isSessionBound) {
```

Comment:
```suggestion
       if (smppSession == null || !smppSession.getSessionState().isBound()) {
```
wouldn't this have been easier?